### PR TITLE
chore(alias): upgrade TypeScript

### PR DESCRIPTION
**Pull request by benmccann**
## Rollup Plugin Name: `alias`

This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [X] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [X] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [X] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### CODE REVIEW

**Comment by benmccann** [comment]
`lint` fails on `master` for me. It seems unrelated to this change

**Review by NotWoods — APPROVED** [review]

**Comment by shellscape** [comment]
Let's focus on one plugin at a time here, especially since there are linting errors abound. #717 can be closed (just did that), and the work merged into this branch. Let's solve the TS linting error here, since it's likely to show up again elsewhere.

All of the other TS upgrade PRs are going to be on hold until we move this one forward. After this one, we'll tackle the next, and so on.

**Comment by benmccann** [comment]
Ok, I've updated this PR to address the lint issues as well. It was trying to use a JavaScript rule on TypeScript code. I fixed that here, but also sent the same change to the base config: https://github.com/rollup/eslint-config-rollup/pull/3

File: .eslintrc.js [hunk 1]
```diff
@@ -32,6 +33,7 @@ module.exports = {
     ],
     'import/no-namespace': 'off',
     'import/no-named-export': 'off',
+    'no-redeclare': 'off',
```
Comment (shellscape) [inline .eslintrc.js:36]: that's not the right fix. see https://github.com/rollup/eslint-config-rollup/pull/3/files#r538998040

Comment (benmccann) [reply .eslintrc.js:36]: this actually does seem like the right fix to me per the conversation on the thread linked in the above comment

**Comment by benmccann** [comment]
@shellscape if you have a chance to take a look at these PRs I think it'd be most helpful to prioritize this one and https://github.com/rollup/plugins/pull/715 in order to fix the lint errors that are happening on `master`

**Pull request merged**